src/EditorFeatures/Core.Wpf/InlineRename/InlineRenameService.cs

```diff
@@ -66,7 +66,7 @@ internal class InlineRenameService : IInlineRenameService
             }
 
             var editorRenameService = document.GetRequiredLanguageService<IEditorInlineRenameService>();
-            var renameInfo = editorRenameService.GetRenameInfoAsync(document, textSpan.Start, cancellationToken).WaitAndGetResult(cancellationToken);
+            var renameInfo = _threadingContext.JoinableTaskFactory.Run(() => editorRenameService.GetRenameInfoAsync(document, textSpan.Start, cancellationToken));
```

Comment:
it woudl be good to have docs here.  this is pretty terrifying.  what about rename is bouncing back to the UI thread in GetRenameInfoAsync?

Reply:
also tagging @jasonmalinowski .

In general, i'm against using JTF.RUn because it may mask an underlying incorrect impl.  i.e. the impl should not ever be going UI, but it is for some reason, and now it is potentially clogging the UI thread without us realizing it.  in that regard, the deadlock is a good thing because it forces us to know where that happens and either redesign accordingly, or we can doc taht this is part of it's behavior so that *any* caller knows tehy'd need to do this if they're blocking on a response from it.

Reply:
➡️ Reverted, and then fixed the underlying issue via 21e13fa